google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClientImpl.java

```diff
@@ -218,7 +218,7 @@ public long executePartitionedUpdate(final Statement stmt, final UpdateOption...
   }
 
   private <T> T runWithSessionRetry(Function<Session, T> callable) {
-    PooledSessionFuture session = getSession();
+    PooledSessionFuture session = getSession(true);
```

Comment:
This should not be statically true, but an input parameter for the method. This method is not a helper for retrying read/write transactions, but for retrying `SessionNotFound` errors that could occur for both read-only and read/write requests. (This happens when the backend has garbage collected or otherwise deleted/invalidated sessions without the client knowing about it)